Objects/regions.c

```diff
@@ -480,4 +480,221 @@ PyObject* _Py_MakeImmutable(PyObject* obj)
     _Py_VPYDBG("<< makeimmutable complete\n\n");
 
     return obj;
-}
\ No newline at end of file
+}
+
+typedef struct RegionObject RegionObject;
+typedef struct RegionMetadata RegionMetadata;
+
+struct RegionMetadata {
+    int lrc;  // Integer field for "local reference count"
+    int osc;  // Integer field for "open subregion count"
+    int is_open;
```

Comment:
Not sure if `open` is required in this interface?  But let's leave for now.

Reply:
Pretty sure it's not since we have the lrc and osc. Tobias and I added it for now to be explicit